src/blocking/wait.rs

```diff
@@ -27,10 +27,9 @@ where
     futures_util::pin_mut!(fut);
 
     loop {
-        match fut.as_mut().poll(&mut cx) {
-            Poll::Ready(Ok(val)) => return Ok(val),
-            Poll::Ready(Err(err)) => return Err(Waited::Inner(err)),
-            Poll::Pending => (), // fallthrough
+        if let Poll::Ready(val) = fut.as_mut().poll(&mut cx) {
+            thread::current().unpark();
```

Comment:
I don't understand what this is meant to do... Wouldn't the current thread already have to be unparked for this code to run at all?

Reply:
Yes, but using `park` consumes a parking token.